servers/rest-services/src/main/java/org/projectnessie/services/rest/RestContentResource.java

```diff
@@ -63,11 +64,17 @@ public RestContentResource(
   }
 
   private ContentApi resource() {
-    return new ContentApiImplWithAuthorization(
-        config,
-        store,
-        accessChecker,
-        securityContext == null ? null : securityContext.getUserPrincipal());
+    return accessChecker != null
```

Comment:
The test case framework is with no access checker, still code was going to ContentApiImplWithAuthorization and providing NPE. Hence this fix.

may be I should separate this change from this PR. For that I have to explore how to add test case for this null access checker scenario.

Reply:
you don't have to set the access checker to null. If authorization is not enabled, then no checks will be performed

Reply:
can you describe what NPE you were hitting? these changes shouldn't be necessary

Reply:
Is it possible to fix the NPE at its root to avoid complicating the logic of classes that delegate to `ContentApiImplWithAuthorization`?

Reply:
It is from the above test framework `AbstractGCTest`, let me narrow down the test cases.

Reply:
In the testcase while mocking the HttpTreeApi.
`HttpTreeApi treeApi = new RestTreeResource(serverConfig, versionStore, null);`
test code is like this, but in the `RestTreeResource` the source code is always goes to `TreeApiImplWithAuthorization` irrespective of access checker and it leads to NPE.

Reply:
Why not use a dummy allow-all access checker in tests?

Reply:
Also, we probably need (new?) tests to validate access checks in the GC workflow too :thinking:  WDYT?

Reply:
Might be preferable to move the dummy access checker impl. to the test jar of `nessie-services` where `AccessChecker` is defined, I guess.